RELEASE_NOTES.txt

```diff
@@ -1,6 +1,120 @@
+* Fixed run ignitevisorcmd in JDK 17 environment.
+* Fixed server-side message parsing for big messages from thin clients.
+* Fixed tasks to sent snapshot files.
+* Fixed the "io.datastorage.StorageSize" metric in case of multiple persistence dataregions.
+* Fixed the command `idle_verify` of the control.sh utility; the command prints and compares full partition counter state.
+* Fixed write empty binary object.
+* Fixed wrong checkpoint locks count calculation after lock timeout (wrong calculation leads to node failure in some cases).
+* Improved REST commands: skip authorization for  'Probe' and 'Version' commands.
+* Improved `--consistency repair` command of the control.sh utility; Read Repair now support arrays and collections as values.
+* Improved `--consistency repair` command of the control.sh utility; Read Repair now support binary keys.
+* Improved `--consistency repair` command of the control.sh utility; atomic caches now repairable by Read Repair.
+* Improved adjustment inline size for fixed size index items.
+* Improved behavior of speed-based throttling when dirty pages ratio is low.
+* Improved index-reader.sh utility: add free size calculation.
+* Improved index-reader.sh utility: analyze and output information about the actual usage of inline space in the index.
+* Improved index-reader.sh utility: exclude repeated stack traces.
+* Improved logging levels for situations when dealing with a client node.
+* Improved registered service name lookup performance and reduced service shutdown times.
+* Improved service name lookup performance and reduced service startup times.
+* Improved speed of the index-reader.sh utility.
+* Lucene library version was updated to 8.11.2.
+* Moved ignite-cloud IP Finder to the Ignite Extensions project.
+* Moved ignite-mesos to the Ignite Extensions project.
+* Moved ignite-spark modules to the Ignite Extensions project.
+* Moved ignite-yarn to the Ignite Extensions project.
+* Moved inigte-aop module to the Ignite Extensions project.
+* Removed deprecated log4j 1.x module.
+* Update Tomcat Servlet API dependency version to 9.0.63.
+* Update jsonpath dependency to 2.7.0.
+* Updated Jackson Databind dependency to 2.12.7 (CVE-2020-36518).
+* Updated Jetty dependency to 9.4.43.
+* Updated Spring dependency to 5.2.22.
+* Updated the MySql connector dependency version (fixes CVE-2021-2471, CVE-2022-21363).
+
+Java thin client:
+* Added AtomicLong.
+* Added automatic binary configuration. Name mapper and compact footer settings will be set according to cluster configuration.
+* Added IgniteSet.
+* Added partition awareness to AtomicLong - send network requests directly to the primary node for the given instance.
+* Added support of IndexQuery feature in Java thin client.
+* Fixed hang on client startup when deprecated TLS version is used.
+* Fixed race condition and potential hang on client close.
+* Fixed an issue that caused addresses are not reloaded from ClientAddressFinder on connection loss.
```

Comment:
89. Fixed an issue that caused addresses not to be reloaded from ClientAddressFinder on connection loss.